TrafficCapture/nettyWireLogging/src/main/java/org/opensearch/migrations/trafficcapture/netty/tracing/WireCaptureContexts.java

```diff
@@ -0,0 +1,284 @@
+                    sourceRequestIndex + 1);
+        }
+    }
+
+    public static class RequestContext
+            extends HttpMessageContext
+            implements IWireCaptureContexts.IRequestContext {
+        public RequestContext(RootWireLoggingContext rootWireLoggingContext,
+                              IConnectionContext enclosingScope,
+                              long sourceRequestIndex) {
+            super(rootWireLoggingContext, enclosingScope, sourceRequestIndex);
+        }
+
+        @Override
+        public IWireCaptureContexts.IWaitingForResponseContext createWaitingForResponseContext() {
+            return new WaitingForResponseContext(getRootInstrumentationScope(), getImmediateEnclosingScope(),
+                    sourceRequestIndex);
+        }
+
+        public static class MetricInstruments extends CommonScopedMetricInstruments {
+            public final LongCounter blockingRequestCounter;
+            public final LongCounter requestsNotOffloadedCounter;
+            public final LongCounter fullyParsedRequestCounter;
+            public final LongCounter bytesReadCounter;
+
+            public MetricInstruments(Meter meter, String activityName) {
+                super(meter, activityName);
+                blockingRequestCounter = meter
+                        .counterBuilder(MetricNames.BLOCKING_REQUEST).setUnit(COUNT_UNITS).build();
+                requestsNotOffloadedCounter = meter
+                        .counterBuilder(MetricNames.CAPTURE_SUPPRESSED).setUnit(COUNT_UNITS).build();
+                fullyParsedRequestCounter = meter
+                        .counterBuilder(MetricNames.FULL_REQUEST).setUnit(COUNT_UNITS).build();
+                bytesReadCounter = meter
+                        .counterBuilder(MetricNames.BYTES_READ).setUnit(BYTES_UNIT).build();
+            }
+        }
+
+        public static @NonNull MetricInstruments makeMetrics(Meter meter) {
+            return new MetricInstruments(meter, ACTIVITY_NAME);
+        }
+
+        @Override
+        public @NonNull MetricInstruments getMetrics() {
+            return getRootInstrumentationScope().getRequestInstruments();
+        }
+
+        @Override
+        public void onBlockingRequest() {
+            meterIncrementEvent(getMetrics().blockingRequestCounter);
+        }
+
+        @Override
+        public void onCaptureSuppressed() {
+            meterIncrementEvent(getMetrics().requestsNotOffloadedCounter);
+        }
+
+        @Override
+        public void onFullyParsedRequest() {
+            meterIncrementEvent(getMetrics().fullyParsedRequestCounter);
+        }
+
+        @Override
+        public void onBytesRead(int size) {
+            meterIncrementEvent(getMetrics().bytesReadCounter, size);
+        }
+    }
+
+    public static class BlockingContext
+            extends HttpMessageContext
+            implements IWireCaptureContexts.IBlockingContext {
+        public static final String ACTIVITY_NAME = "blocked";
```

Comment:
Should we align this to a ActivityNames class like the replayer code?

Reply:
Done - thanks